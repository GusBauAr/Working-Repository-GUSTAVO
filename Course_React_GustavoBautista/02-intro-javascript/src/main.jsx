


console.log('Hola mundo de Gus')