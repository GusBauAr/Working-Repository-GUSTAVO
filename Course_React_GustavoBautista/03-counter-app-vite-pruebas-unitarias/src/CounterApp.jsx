import {useState} from 'react'; //useState - Hook
import PropTypes from 'prop-types';
import React from 'react'



export const CounterApp = ({ value }) => {
    //nuestro primer hooks

    const [ counter, sentCounter ] = useState( value )
    
    const handleAdd  = () =>{
        // console.log(event)
        sentCounter( counter + 1);
    }
    const menos =() =>{
        sentCounter(counter - 1);
    }
    const reset = ()=>{
        sentCounter(value)
    }


  return (
    <>
        <h1>CounterApp</h1>
        <h2> { counter } </h2>

        <button onClick={handleAdd}>+1</button>
        <button onClick={menos}> -1</button>
        <button onClick={reset}> Reset</button>
    </>
    
  )
}

CounterApp.protoTypes ={
    value: PropTypes.number.isRequired
}
