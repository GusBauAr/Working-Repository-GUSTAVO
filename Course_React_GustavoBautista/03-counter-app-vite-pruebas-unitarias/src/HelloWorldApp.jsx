// function App() {
//     return (<h1>Hola Mundo</h1>);
// }

// export default App;

export const HelloWorldApp = () => {


    return (
      <h1>Hello World App</h1>
    );
}
  
  // FirstApp  <h1>Fernando</h1>





































// //!creamos nuestra funcion comun 
// // export function HelloWorldApp () {
// //     return(<h1>Hola mundo GUSTAVOooooo</h1>);
// // }
// //!creamosuna funcion de flecha
// export const HelloWorldApp = () =>{

//     return(
//         <h1>HOLA MUNDIO DE GUSTAVO</h1>
//     );
// }

// //!hay una manra que puede ser mas simplificado
// // export const HelloWorldApp = () => <h1>Hola mundo de gustavo</h1>
