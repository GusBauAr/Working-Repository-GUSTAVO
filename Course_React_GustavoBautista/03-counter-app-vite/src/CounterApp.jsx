
import PropTypes from 'prop-types';

import React from 'react'

export const CounterApp = ({value}) => {
  return (
    <>
        <div>CounterApp</div>
        <h2>{value}</h2>
    </>
    
  )
}

CounterApp.protoTypes ={
    value: PropTypes.number.isRequired
}
