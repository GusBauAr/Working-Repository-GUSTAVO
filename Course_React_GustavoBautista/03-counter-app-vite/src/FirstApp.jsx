//objeto de las props
export const FirstApp =({title, subTitle}) =>{

    return(
        //esto es un sinomino de un fracmento, es un agrupador de otros elentos htmlen este caso tenemos neustro nodo padre (<></>)
        //y colocamos el newMessage en el H1 para que salga Gustavo
        <> 
        <h1>{title}</h1> 
        <p>{ subTitle}</p>
        </>

    );
}


//!IMPORTANTE
//!comunicacion entre componentes props
//todo lo que definamos en el main.jsx: 
// <React.StrictMode>
//        <FirstApp title={"Hola , soy GUSTAVO"}/>
//    </React.StrictMode>
//
//?son props que estamos  mandando del componente padre al componenete hijo que es mi aplicacion FirstApp.jsx













