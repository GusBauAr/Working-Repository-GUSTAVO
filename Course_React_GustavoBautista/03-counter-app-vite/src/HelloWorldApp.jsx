//!creamos nuestra funcion comun 
// export function HelloWorldApp () {
//     return(<h1>Hola mundo GUSTAVOooooo</h1>);
// }
//!creamosuna funcion de flecha
export const HelloWorldApp = () =>{

    return(
        <h1>HOLA MUNDIO DE GUSTAVO</h1>
    );
}

//!hay una manra que puede ser mas simplificado
// export const HelloWorldApp = () => <h1>Hola mundo de gustavo</h1>
