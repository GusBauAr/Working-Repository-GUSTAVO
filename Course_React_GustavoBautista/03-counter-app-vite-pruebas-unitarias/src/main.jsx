import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';

import { FirstApp } from './FirstApp';
// import { HelloWorldApp } from './HelloWorldApp';
import './styles.css';



ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        {/* <CounterApp value={ 20 } /> */}
        <FirstApp title="Hola, Soy Vegeta" />
    </React.StrictMode>
);



























// // se necesita esta importanciones para renderizar mi aplicacion..........................................
// import React from 'react';
// import ReactDOM from 'react-dom/client';
// // import { HelloWorldApp } from './HelloWorldApp';
// import { FirstApp } from './FirstApp';
// // .........................................
// import './styles.css'
// // import { CounterApp } from './CounterApp';


// ReactDOM.createRoot(document.getElementById('root')).render(
//     <React.StrictMode>
//         {/* <CounterApp value={20}/> */}
//         <FirstApp/>
//     </React.StrictMode>
// );
















