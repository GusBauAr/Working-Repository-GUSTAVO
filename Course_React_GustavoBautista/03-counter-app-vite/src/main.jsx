// se necesita esta importanciones para renderizar mi aplicacion..........................................
import React from 'react';
import ReactDOM from 'react-dom/client';
// import { HelloWorldApp } from './HelloWorldApp';
import { FirstApp } from './FirstApp';
// .........................................
import './styles.css'


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <FirstApp title='HOLA SOY GUSTAVO' subTitle={123}/>
    </React.StrictMode>
);
















