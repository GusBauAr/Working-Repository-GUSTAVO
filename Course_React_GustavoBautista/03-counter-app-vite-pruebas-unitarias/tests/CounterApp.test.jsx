import { render, screen } from "@testing-library/react"
import { CounterApp } from "../src/CounterApp"


describe('pruebas  en el CounterApp.jsx', () => { 

    const initialValue = 10;

    test('debe de hacer match con el snapshot', () => { 

        const{ conainer } = render(<CounterApp value={ initialValue }/>);

        expect( conainer ).toMatchSnapshot();
    });


    test('debe de mostrar el valorinicial de 100 <CounterApp value={100}/>', () => { 
        render(<CounterApp value={100}/>);
        expect(screen.getByText('100') ).toBeTruthy();
        
        expect (screen.getByRole('heading',{level:2}).innerHTML).toContain('100');
    })

})