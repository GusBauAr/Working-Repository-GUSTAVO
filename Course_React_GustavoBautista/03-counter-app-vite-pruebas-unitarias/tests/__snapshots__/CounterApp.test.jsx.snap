// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas  en el CounterApp.jsx debe de hacer match con el snapshot 1`] = `undefined`;
